pkg/ddc/juicefs/status.go

```diff
@@ -16,7 +16,98 @@ limitations under the License.
 
 package juicefs
 
-func (j JuiceFSEngine) CheckAndUpdateRuntimeStatus() (ready bool, err error) {
-	// todo
-	return true, nil
+import (
+	"context"
+	"reflect"
+	"time"
+
+	data "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"github.com/fluid-cloudnative/fluid/pkg/utils"
+	"k8s.io/client-go/util/retry"
+)
+
+// CheckAndUpdateRuntimeStatus checks the related runtime status and updates it.
+func (j *JuiceFSEngine) CheckAndUpdateRuntimeStatus() (ready bool, err error) {
+
+	var (
+		workerReady bool
+		workerName  string = j.getWorkerDaemonsetName()
+		namespace   string = j.namespace
+	)
+
+	// 1. Worker should be ready
+	workers, err := j.getDaemonset(workerName, namespace)
+	if err != nil {
+		return ready, err
+	}
+
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		runtime, err := j.getRuntime()
+		if err != nil {
+			return err
+		}
+
+		runtimeToUpdate := runtime.DeepCopy()
+		if reflect.DeepEqual(runtime.Status, runtimeToUpdate.Status) {
+			j.Log.V(1).Info("The runtime is equal after deepcopy")
+		}
+
+		states, err := j.queryCacheStatus()
+		if err != nil {
+			return err
+		}
+
+		// 0. Update the cache status
+		if len(runtime.Status.CacheStates) == 0 {
+			runtimeToUpdate.Status.CacheStates = map[common.CacheStateName]string{}
+		}
+
+		runtimeToUpdate.Status.CacheStates[common.CacheCapacity] = states.cacheCapacity
```

Comment:
Data cache in each JuiceFS fuse pod may be not equal, we do not count cache status in first version.